texar/torch/modules/tokenizers/pretrained_roberta_tokenizer.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Pre-trained RoBERTa tokenizer.
+"""
+
+from typing import Any, Dict
+
+from texar.torch.modules.tokenizers.pretrained_gpt2_tokenizer import \
+    GPT2Tokenizer
+
+__all__ = [
+    'RoBERTaTokenizer',
+]
+
+_GPT2_PATH = "https://storage.googleapis.com/gpt-2/models/"
+_CHECKPOINT_FILES = ["encoder.json", "vocab.bpe"]
+
+
+class RoBERTaTokenizer(GPT2Tokenizer):
+    r"""Pre-trained RoBERTa Tokenizer.
+
+    Args:
+        pretrained_model_name (optional): a `str`, the name of
+            pre-trained model (e.g., `roberta-base`). Please refer to
+            :class:`~texar.torch.modules.pretrained.PretrainedRoBERTaMixin` for
+            all supported models.
+            If None, the model name in :attr:hparams is used.
+        cache_dir (optional): the path to a folder in which the
+            pre-trained models will be cached. If `None` (default),
+            a default directory will be used.
+        hparams (dict or HParams, optional): Hyperparameters. Missing
+            hyperparameter will be set to default values. See
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    _MODEL2URL = {
+        'roberta-base': [_GPT2_PATH + f"117M/{file}"
+                         for file in _CHECKPOINT_FILES],
```

Comment:
Does RoBERTa use the exact same tokenizer as GPT2? If so, we should probably document it, otherwise it's going to cause confusion.

Reply:
Yes, `RoBERTa` applies the exact same tokenizer as `GPT2`.

Reply:
Did RoBERTa release their own tokenizer files? If so I guess we can first check if RoBERTa files exist, and resort to download the GPT2 one if not.

Reply:
To the best of my knowledge, they did not release their own tokenization files. In their [example](https://github.com/pytorch/fairseq/blob/master/examples/roberta/multiprocessing_bpe_encoder.py#L23), they directly refer to `GPT2`'s tokenization files.

Reply:
I see, but let's also use the same URL just in case.